packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -148,21 +151,36 @@ export const useTreeItem2 = <TPlugins extends DefaultTreeViewPlugins = DefaultTr
       onBlur: createRootHandleBlur(externalEventHandlers),
       onKeyDown: createRootHandleKeyDown(externalEventHandlers),
     };
+
+    if (indentationAtItemLevel) {
```

Comment:
To avoid any breaking change, I'm only adding those props when the experimental feature is enabled.
When not using the experimental feature, the props passed to each slot remain unchanged.